docs/guides/about_the_dangerfile.html.md

```diff
@@ -0,0 +1,91 @@
+---
+title: About the Dangerfile
+subtitle: The Dangerfile
+layout: guide_kt
+order: 1
+blurb: Step two on using Danger in your app, how to work locally and nuances around working with files.
+---
+
+## Writing your Dangerfile
+
+All of Danger is built in Kotlin, aims to have 100% in-line docs. This means a lot of your exploration can be done inside Intellij Idea. 
+This document aims to give you some high level knowledge on how to work on your `Dangerfile.df.kts`.
+
+## Working on your Dangerfile
+
+There are two ways to locally work on your Dangerfile. These both rely on using an external API locally, so you may hit
+their API rate-limits or need to have authenticated request for private repos. In which case you can use an access token
+to do authenticated requests by exposing a token to Danger.
+
+```sh
+export DANGER_GITHUB_API_TOKEN='xxxx'
+
+# or for BitBucket by username and password
+export DANGER_BITBUCKETSERVER_HOST='xxxx' DANGER_BITBUCKETSERVER_USERNAME='yyyy' DANGER_BITBUCKETSERVER_PASSWORD='zzzz'
+
+# or for BitBucket by username and personal access token
+export DANGER_BITBUCKETSERVER_HOST='xxxx' DANGER_BITBUCKETSERVER_USERNAME='yyyy' DANGER_BITBUCKETSERVER_TOKEN='zzzz'
+```
+
+Then the danger CLI will use authenticated API calls, which don't get hit by API limits.
+
+### Using danger pr
+
+The command `danger-kotlin pr` expects an argument of a PR url, e.g:
+
+```sh
+danger-kotlin pr https://github.com/danger/swift/pull/95
+```
+
+This will use your local `Dangerfile.swift` against the metadata of the linked PR. Danger will then output the results
```

Comment:
Dangerfile.df.kts

Reply:
What is the `.df.` for?

Reply:
Is a pattern used to specify the kind of Kotlin script we are using.. the pure simple Kotlin script is not enough for our purposes, then we defined a custom Kotlin script with extension .df.kts where .df means DangerFile. The script definition that you can find in our codebase imports automatically the danger framework with our features:
- maven deps
- plugins
😎😎😎